### PR TITLE
Pretty scheduling diagnostics

### LINKED ISSUE #1583: Scheduling checks red/green bars
This is mostly a note to myself, suggested by @gkanwar.  It would be cool if the scheduling checks page loaded things in the background, and then if the scheduling check had no content (i.e. everything was correct) it would turn the bar green, or something like that.

### LINKED ISSUE #2764: Reformat/prettify scheduling diagnostics
The scheduling diagnostics page is dense and difficult to read.
image

In particular, I propose

- adding more obvious expand/collapse button
- adding more space between each diagnostic
- sorting them into categories
- making more use of the horizontal space to avoid having the long single column
- changing the list of users (e.g. for the "Admins teaching per timeslot" check) to use names in addition to usernames and not list them with quotes and brackets

### PR DESCRIPTION
**Pull request by nrjean**
Fixes #2764 

Note on 2764 - issue specifically mentioned with was a file missing from Granite Hills specific site. That was since been fixed.

This PR does several things:

Firstly - regarding #1583, all the scheduling diagnostics auto load on page load. While they load a yellow "loading" bar appears. Once the load is complete either a green bar for loading success appears or a red bar for loading failed appears. Note that a failed loading can be simulated by stopping the server mid-load.

Graphically regarding 2764 there are a bunch of changes.
Everything is a little prettier, rounded corners and the like
Added a new button called Open/Close which does exactly what is says
Changed the swirly symbol to say reload which reloads and individual scheduling diagnostic
Refresh still works that same way
Added hard coded help text to the top of the page
Reordered the diagnostics to ground them by catagory

### CODE REVIEW

**Comment by nrjean** [comment]
@kkbrum I've addressed all your save but the load status one. 
As discussed, I think the load status would require a large enough rewrite of the scheduling check javascript that it can be left to its own PR. 

All diagnostics that have usernames also have name names as well. I also spend the time to take every diagnostic that appeared in quotes and fixed it so only the text appeared.

Capitalization is now consistent across all the headers and the headers now have a bit of padding so everything is not as squishy

I made the tables look a bit more modern. Also added the option to add a header to a list based diagnostic, this keeps the formatting consistent across the checks.

File: esp/esp/program/modules/handlers/schedulingcheckmodule.py [hunk 1]
```diff
@@ -152,10 +153,10 @@ def run_diagnostics(self, diagnostics=None):
          ('special_classroom_types', 'Special Classroom Types'),
```
Comment (kkbrum) [inline esp/esp/program/modules/handlers/schedulingcheckmodule.py:153]:
```suggestion
         ('special_classroom_types', 'Special classroom types'),
```

File: esp/esp/program/modules/handlers/schedulingcheckmodule.py [hunk 2]
```diff
@@ -165,26 +166,14 @@ def run_diagnostics(self, diagnostics=None):
          ('teachers_teaching_two_classes_same_time', 'Teachers teaching two classes at once'),
          ('teachers_who_like_running', 'Teachers who like running'),
          ('hungry_teachers', 'Hungry Teachers'),
-         ('hosed_teachers', 'Hosed Teachers'),
+         ('inflexible_teachers', 'Teachers with Limited Flexibility'),
```
Comment (kkbrum) [inline esp/esp/program/modules/handlers/schedulingcheckmodule.py:169]:
```suggestion
         ('inflexible_teachers', 'Teachers with limited flexibility'),
```

File: esp/esp/program/modules/handlers/schedulingcheckmodule.py [hunk 3]
```diff
@@ -146,29 +147,33 @@ def run_diagnostics(self, diagnostics=None):
...
+         ('room_capacity_mismatch', 'Class max size/room max size mismatches'),
+         #Class Diagnostiscs
+         ('wrong_classroom_type', 'Classes in wrong classroom type'),
+         ('classes_missing_resources', 'Unfulfilled resource requests'),
+         ('missing_resources_by_hour', 'Unfulfilled resource requests by hour'),
+         ('incompletely_scheduled_classes', 'Classes not completely scheduled or with gaps'),
+         ('classes_which_cover_lunch', 'Classes which are scheduled over lunch'),
+         ('classes_wrong_length', 'Classes which are the wrong length'),
+         ('no_overlap_classes', "Classes which shouldn't overlap"),
+         ('unapproved_scheduled_classes', 'Classes which are scheduled but aren\'t approved'),
```
Comment (kkbrum) [inline esp/esp/program/modules/handlers/schedulingcheckmodule.py:163]:
```suggestion
         ('unapproved_scheduled_classes', 'Classes which are scheduled but not approved'),
```

This is just to make it slightly shorter so the "Reset" button lines up at least in this theme 🙃 

image

image

File: esp/esp/program/modules/handlers/schedulingcheckmodule.py [hunk 4]
```diff
@@ -146,29 +147,33 @@ def run_diagnostics(self, diagnostics=None):
...
+         #Teacher Diagnostics
+         ('teachers_unavailable', "Teachers teaching when they aren't available"),
+         ('teachers_teaching_two_classes_same_time', 'Teachers teaching two classes at once'),
+         ('teachers_who_like_running', 'Teachers who like running'),
+         ('hungry_teachers', 'Hungry Teachers'),
```
Comment (kkbrum) [inline esp/esp/program/modules/handlers/schedulingcheckmodule.py:168]:
```suggestion
         ('hungry_teachers', 'Hungry teachers'),
```

**Comment by nrjean** [comment]
@kkbrum 
I made the capitalization changes as well as fixed the issue with the spacing.

I edited this on the fly so wasn't able to test it - but the change was simply to change str1 from " " to ", ". In the .join command which creates those strings str1 as defined is the delimited so that change should have simple added a comma delimiter without changing anything else.

**Pull request merged**